datajoint/autopopulate.py

```diff
@@ -39,31 +39,41 @@ def target(self):
 
     def populate(self, restriction=None, suppress_errors=False, reserve_jobs=False):
         """
-        rel.populate() calls rel._make_tuples(key) for every primary key in self.pop_rel
+        rel.populate() calls rel._make_tuples(key) for every primary key in self.populate_relation
         for which there is not already a tuple in rel.
+
+        :param restriction: restriction on rel.populate_relation - target
+        :param suppress_errors: suppresses error if true
+        :param reserve_jobs: currently not implemented
         """
-        assert not reserve_jobs, NotImplemented   # issue #5
+        assert not reserve_jobs, NotImplemented  # issue #5
+
         error_list = [] if suppress_errors else None
+
         if not isinstance(self.populate_relation, RelationalOperand):
             raise DataJointError('Invalid populate_relation value')
-        self.conn._cancel_transaction()  # rollback previous transaction, if any
+
         unpopulated = (self.populate_relation - self.target) & restriction
+
         for key in unpopulated.project():
-            self.conn._start_transaction()
-            if key in self.target:  # already populated
-                self.conn._cancel_transaction()
-            else:
-                logger.info('Populating: ' + str(key))
-                try:
-                    self._make_tuples(key)
-                except Exception as error:
-                    self.conn._cancel_transaction()
-                    if not suppress_errors:
-                        raise
-                    else:
-                        print(error)
-                        error_list.append((key, error))
+            try:
+                while True:
+                    try:
+                        with self.conn.transaction():
+                            if not key in self.target:  # already populated
+                                logger.info('Populating: ' + str(key))
+                                self._make_tuples(dict(key))
+                        break
+                    except TransactionError as tr_err:
+                        if suppress_errors:
+                            error_list.append((key,tr_err))
```

Comment:
No matter what solution we agree on, I think it should have two features:
* There should be a way to check whether TransactionErrors have been raised (and dealt with). I think this is important information for the user (I guess no one likes to compute something again just because he forgot to declare a table) and I'd like to have that feature for testing. 
* We should make it very clear that only unhandled errors are part of the error list. Otherwise I think the current behaviour is unexpected. That's why I decided to add the errors to the list. That said, if errors are not suppressed, then the user does not see the error at all. I don't like that part of the current solution. I'd actually favour always returning an error list with the possibility that it is empty.